src/common/librlist/rnode.c

```diff
@@ -197,7 +198,17 @@ int rnode_add (struct rnode *orig, struct rnode *n)
             return -1;
         c = zhashx_next (n->children);
     }
-    return 0;
+    if (n->properties) {
+        zlistx_t *l = zhashx_keys (n->properties);
```

Comment:
Need to check `zhashx_keys()` for NULL return here.

Reply:
Thanks! I addressed this and force pushed the result.